auth0/src/main/java/com/auth0/android/provider/IdTokenVerifier.java

```diff
@@ -0,0 +1,149 @@
+        cal.setTime(token.getIssuedAt());
+        cal.add(Calendar.SECOND, -1 * clockSkew);
+        Date iatDate = cal.getTime();
+
+        if (now.before(iatDate)) {
+            throw new TokenValidationException(String.format("Issued At (iat) claim error in the ID token; current time (%d) is before issued at time (%d)", now.getTime() / 1000, iatDate.getTime() / 1000));
+        }
+
+
+        if (verifyOptions.nonce != null) {
+            String nonceClaim = token.getClaim(NONCE_CLAIM).asString();
+            if (isEmpty(nonceClaim)) {
+                throw new TokenValidationException("Nonce (nonce) claim must be a string present in the ID token");
+            }
+            if (!verifyOptions.nonce.equals(nonceClaim)) {
+                throw new TokenValidationException(String.format("Nonce (nonce) claim mismatch in the ID token; expected \"%s\", found \"%s\"", verifyOptions.nonce, nonceClaim));
+            }
+        }
+
+        if (audience.size() > 1) {
+            String azpClaim = token.getClaim(AZP_CLAIM).asString();
+            if (isEmpty(azpClaim)) {
+                throw new TokenValidationException("Authorized Party (azp) claim must be a string present in the ID token when Audience (aud) claim has multiple values");
+            }
+            if (!verifyOptions.audience.equals(azpClaim)) {
+                throw new TokenValidationException(String.format("Authorized Party (azp) claim mismatch in the ID token; expected \"%s\", found \"%s\"", verifyOptions.audience, azpClaim));
+            }
+        }
+
+        if (verifyOptions.maxAge != null) {
+            Date authTime = token.getClaim(AUTH_TIME_CLAIM).asDate();
+            if (authTime == null) {
+                throw new TokenValidationException("Authentication Time (auth_time) claim must be a number present in the ID token when Max Age (max_age) is specified");
+            }
+
+            cal.setTime(authTime);
+            cal.add(Calendar.SECOND, verifyOptions.maxAge);
+            cal.add(Calendar.SECOND, clockSkew);
+            Date authTimeDate = cal.getTime();
+
+            if (now.after(authTimeDate)) {
+                throw new TokenValidationException(String.format("Authentication Time (auth_time) claim in the ID token indicates that too much time has passed since the last end-user authentication. Current time (%d) is after last auth at (%d)", now.getTime() / 1000, authTimeDate.getTime() / 1000));
+            }
+        }
+    }
+
+    static class Options {
+        final String issuer;
+        final String audience;
+        final SignatureVerifier verifier;
+        String nonce;
+        private Integer maxAge;
+        Integer clockSkew;
+        Date clock;
+
+        Options(@NonNull String issuer, @NonNull String audience, @NonNull SignatureVerifier verifier) {
+            this.issuer = issuer;
+            this.audience = audience;
+            this.verifier = verifier;
+        }
+
+        void setNonce(@Nullable String nonce) {
```

Comment:
we use the setters for setting the values, but then here we access them directly. Should we remove the setters as well?

Reply:
Not sure exactly what you're asking but seems fine either way.